phoenix-core/src/main/java/org/apache/phoenix/parse/PSchema.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.parse;
+
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.phoenix.coprocessor.generated.PSchemaProtos;
+import org.apache.phoenix.schema.PMetaDataEntity;
+import org.apache.phoenix.schema.PName;
+import org.apache.phoenix.schema.PNameFactory;
+import org.apache.phoenix.schema.PTableKey;
+import org.apache.phoenix.util.SchemaUtil;
+import org.apache.phoenix.util.SizedUtil;
+
+public class PSchema implements PMetaDataEntity {
+
+    private final PName schemaName;
+    private PTableKey schemaKey;
+    private long timeStamp;
+    private int estimatedSize;
+
+    public PSchema(long timeStamp) { // For index delete marker
+        this.timeStamp = timeStamp;
+        this.schemaName = null;
+    }
+
+    public PSchema(String schemaName) {
+        this(schemaName, HConstants.LATEST_TIMESTAMP);
+    }
+
+    public PSchema(String schemaName, long timeStamp) {
+        this.schemaName = PNameFactory.newName(SchemaUtil.normalizeIdentifier(schemaName));
+        this.schemaKey = new PTableKey(null, this.schemaName.getString());
```

Comment:
Thinking about this a little bit more, is it possible for tenant views on the same multi-tenant table to have different schema/namespaces? If yes, how would you differentiate these schemas considering the tenantId is being set as null here. At a minimum, it would be ideal to have some testing around this scenario.